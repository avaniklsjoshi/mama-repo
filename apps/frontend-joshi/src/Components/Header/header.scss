@import "../../CommonStyles/placeholders.scss";
@import "../../CommonStyles/variables.scss";

.header-container {
  @extend %center;
  padding: 20px;
}
.header-title {
  margin-right: auto;
  font-size: $appTitleFontSize;
  cursor: pointer;
  align-self: flex-end;
  transition: all 0.2s ease-in;
  &:hover {
    color: #fdec6e;
    text-shadow: 0px 0px 5px #fdec6e;
    transition: all 0.2s ease-in;
  }
}
.header-accessibilities {
  @extend %center;
  flex-direction: column;
  div {
    color: inherit;
    margin-left: auto;
    max-height: 46px;
  }
  svg {
    color: inherit;
  }
  .header-lang-picker {
    min-width: 135px;
    display: flex;
    justify-content: flex-end;
  }
}
.feature-language {
  display: flex;
  align-items: baseline;
}
.language-selectbox {
  margin: 0 !important;
  fieldSet {
    border: none;
  }
}
#language-select {
  margin: 0;
}
.theme-picker {
  img {
    height: 24px;
    width: 24px;
  }
  button {
    border: none;
    border-radius: 24px !important;
    &.Mui-selected,
    &.Mui-selected:hover {
      background: none;
      .img-back {
        box-shadow: 1px 1.5px 23px red;
        height: 15px;
        width: 15px;
        border-radius: 18px;
        position: absolute;
      }
    }
    .img-back {
      position: absolute;
    }
  }
  button:hover {
    box-shadow: 1px 1.5px 8px red;
    border-radius: 24px;
  }
}
.header-navigator {
  cursor: pointer;
}
