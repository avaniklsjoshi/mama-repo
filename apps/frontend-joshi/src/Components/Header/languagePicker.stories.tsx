import { ComponentMeta, ComponentStory } from "@storybook/react";
import { userEvent, within } from "@storybook/testing-library";

import LanguagePicker from "./languagePicker";

export default {
  component: LanguagePicker,
  title: "Header/Language Picker",
  parameters: {
    layout: "fullscreen"
  }
} as ComponentMeta<typeof LanguagePicker>;
const Template: ComponentStory<typeof LanguagePicker> = () => (
  <LanguagePicker />
);

export const SelectClick = Template.bind({});

// More on interaction testing: https://storybook.js.org/docs/react/writing-tests/interaction-testing
SelectClick.play = async ({ canvasElement }) => {
  const canvas = within(canvasElement);
  const loginButton = await canvas.getByRole("select", { name: /English/i });
  await userEvent.click(loginButton);
};
