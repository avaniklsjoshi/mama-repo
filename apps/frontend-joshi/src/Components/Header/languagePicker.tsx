import { MenuItem, Select } from "@mui/material";
import { useTranslation } from "react-i18next";
import {
  DEFAULT_LANGUAGE,
  LANGUAGE_NAMES,
  LANGUAGE_NAMES_KEY
} from "./../../Configs/Constants/const";

export default function LanguagePicker() {
  const { i18n } = useTranslation();

  const changeLanguageHandler = (e: any) => {
    const languageValue = e.target.value;
    i18n.changeLanguage(languageValue);
  };

  return (
    <Select
      id="language-select"
      defaultValue={DEFAULT_LANGUAGE}
      onChange={changeLanguageHandler}
      className="language-selectbox"
    >
      <MenuItem value={LANGUAGE_NAMES_KEY.HINDI}>
        {LANGUAGE_NAMES.HINDI}
      </MenuItem>
      <MenuItem value={LANGUAGE_NAMES_KEY.ENGLISH}>
        {LANGUAGE_NAMES.ENGLISH}
      </MenuItem>
      <MenuItem value={LANGUAGE_NAMES_KEY.GERMAN}>
        {LANGUAGE_NAMES.GERMAN}
      </MenuItem>
    </Select>
  );
}
