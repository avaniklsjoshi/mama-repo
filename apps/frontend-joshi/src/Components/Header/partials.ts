import styled from "styled-components";

export const HeaderContainer: any = styled.div`
  background: ${(props) => props.theme.header.background};
  color: ${(props) => props.theme.header.text};
  box-shadow: 0 -3px 18px 13px grey;
  backdrop-filter: saturate(50%) blur(8px);
  margin-bottom: 30px;
  position: fixed;
    top: 0;
    width: 100%;
}
`;
