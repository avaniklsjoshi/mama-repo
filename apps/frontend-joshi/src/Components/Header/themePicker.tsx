import { ToggleButton, ToggleButtonGroup } from "@mui/material";
import { useCallback, useContext, useEffect, useState } from "react";
import {
  DEFAULT_THEME,
  LOCAL_STORAGE_THEME_KEY,
  THEME_ARRAY
} from "../../Configs/Constants/const";
import { IThemeConfig } from "../../Theme/themeConfig";
import { PlaceThemeContext } from "../../Theme/themeContext";
import { useTheme } from "../../Theme/useTheme";
import { setToLS } from "../../Utils/localStorage";

export default function ThemePicker() {
  const themeContext = useContext(PlaceThemeContext);
  const [alignment, setAlignment] = useState(DEFAULT_THEME);
  const { theme } = useTheme();

  const themeSlection = useCallback(
    (type: any) => {
      const selectedTheme: IThemeConfig = THEME_ARRAY.filter((theme) => {
        return theme.themeName === type.currentTarget.value;
      })[0];
      themeContext.setNewTheme(selectedTheme.themeValue);
      setToLS(LOCAL_STORAGE_THEME_KEY, selectedTheme.themeValue);
      setAlignment(selectedTheme.themeValue);
    },
    [themeContext]
  );

  useEffect(() => {
    if (theme) {
      setAlignment(() => theme);
    }
  }, [theme]);

  return (
    <div className="theme-picker">
      <ToggleButtonGroup
        color="primary"
        exclusive
        onChange={themeSlection}
        value={alignment.themeType}
      >
        {THEME_ARRAY.map((theme: IThemeConfig, index: number) => (
          <ToggleButton value={theme.themeName} key={index}>
            <img src={theme.themeIcon} alt="theme logo" className="box-image" />
            <div className="img-back"></div>
          </ToggleButton>
        ))}
      </ToggleButtonGroup>
    </div>
  );
}
