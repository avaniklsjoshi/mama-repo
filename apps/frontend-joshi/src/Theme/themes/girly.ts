export default {
  themeType: "girly",
  colors: {
    background: "#f1b7ea",
    text: "#fff"
  },
  header: {
    background: "#000",
    text: "#fff"
  },
  footer: {
    background: "#262726",
    text: "#fff"
  },
  section: {
    background: "#008F7A",
    text: "#fff"
  }
};
