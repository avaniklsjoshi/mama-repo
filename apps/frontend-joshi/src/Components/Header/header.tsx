import { useTranslation } from "react-i18next";
import { useNavigate } from "react-router-dom";
import LanguagePicker from "./languagePicker";
import * as SC from "./partials";
import ThemePicker from "./themePicker";
import "./header.scss";
import { ROUTES, ROUTES_NAME_KEY } from "../../Configs/Constants/const";

export default function Header() {
  const { t } = useTranslation();
  const navigate = useNavigate();

  const navigateToFeatures = () => {
    navigate(ROUTES.FEATURES);
  };
  const navigateHome = () => {
    navigate(ROUTES.HOME);
  };

  return (
    <SC.HeaderContainer>
      <div className="header-container">
        <div className="header-title" onClick={navigateHome} aria-hidden="true">
          {t("avani")}
        </div>
        <div className="header-accessibilities">
          <div>
            <ThemePicker />
          </div>
          <div className="feature-language">
            <div
              className="header-navigator"
              onClick={navigateToFeatures}
              aria-hidden="true"
            >
              {t(ROUTES_NAME_KEY.FEATURES)}
            </div>
            <div className="header-lang-picker">
              <LanguagePicker />
            </div>
          </div>
        </div>
      </div>
    </SC.HeaderContainer>
  );
}
