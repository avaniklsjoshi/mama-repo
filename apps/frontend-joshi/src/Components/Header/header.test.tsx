/* eslint-disable testing-library/no-node-access */
/* eslint-disable testing-library/no-container */ import {
  act,
  fireEvent,
  render,
  screen
} from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import { ThemeProvider } from "styled-components";
import Header from "./header";
import LanguagePicker from "./languagePicker";
import ThemePicker from "./themePicker";
import girly from "../../Theme/themes/girly";

jest.mock("react-i18next", () => ({
  useTranslation: () => ({ t: (key: any) => key })
}));

describe("Header", () => {
  it("renders Header", () => {
    const { container } = render(
      <ThemeProvider theme={girly}>
        <Router>
          <Header />
        </Router>
      </ThemeProvider>
    );
    expect(container).toMatchSnapshot();
    const featureButton = screen.getByTestId("feature-button");
    fireEvent.click(featureButton);

    const homeButton = screen.getByTestId("home-button");
    fireEvent.click(homeButton);
    expect(screen.getByText("avani")).toBeInTheDocument();
  });
  it("renders Header ThemePicker", () => {
    const { container } = render(<ThemePicker />);
    expect(container).toMatchSnapshot();

    const selectTheme = screen.getByTestId("select-theme");
    fireEvent.change(selectTheme);
  });
  it("renders Header LanguagePicker", () => {
    const { container } = render(<LanguagePicker />);
    expect(container).toMatchSnapshot();

    const selectLang = screen.getByTestId("language-select");
    fireEvent.change(selectLang);
  });
});
