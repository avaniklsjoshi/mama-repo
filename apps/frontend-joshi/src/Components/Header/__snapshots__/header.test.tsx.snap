// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders Header 1`] = `
<div>
  <div
    class="sc-bczRLJ hMWBre"
  >
    <div
      class="header-container"
    >
      <div
        aria-hidden="true"
        class="header-title"
        data-testid="home-button"
      >
        <div
          class="sc-gsnTZi kmpfjj"
        >
          avani
        </div>
      </div>
      <div
        class="header-accessibilities"
      >
        <div
          class="theme-picker"
        >
          <div
            class="MuiToggleButtonGroup-root css-1gjgmky-MuiToggleButtonGroup-root"
            data-testid="select-theme"
            role="group"
          >
            <button
              aria-pressed="false"
              class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-primary MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-d9c359-MuiButtonBase-root-MuiToggleButton-root"
              tabindex="0"
              type="button"
              value="girly"
            >
              <img
                alt="theme logo"
                class="box-image"
                src="colorMode.png"
              />
              <div
                class="img-back"
              />
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-pressed="false"
              class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-primary MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-d9c359-MuiButtonBase-root-MuiToggleButton-root"
              tabindex="0"
              type="button"
              value="light"
            >
              <img
                alt="theme logo"
                class="box-image"
                src="dayMode.png"
              />
              <div
                class="img-back"
              />
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-pressed="true"
              class="MuiButtonBase-root MuiToggleButton-root Mui-selected MuiToggleButton-sizeMedium MuiToggleButton-primary MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-d9c359-MuiButtonBase-root-MuiToggleButton-root"
              tabindex="0"
              type="button"
              value="dark"
            >
              <img
                alt="theme logo"
                class="box-image"
                src="nightMode.png"
              />
              <div
                class="img-back"
              />
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <div
          class="feature-language"
        >
          <div
            aria-hidden="true"
            class="header-navigator"
            data-testid="feature-button"
          >
            featuresRoute
          </div>
          <div
            class="header-lang-picker"
          >
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary language-selectbox css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
              data-testid="language-select"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="language-select"
                class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                id="language-select"
                role="button"
                tabindex="0"
              >
                English
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                tabindex="-1"
                value="en"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-hdw1oc"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Header renders Header LanguagePicker 1`] = `
<div>
  <div
    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary language-selectbox css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
    data-testid="language-select"
  >
    <div
      aria-expanded="false"
      aria-haspopup="listbox"
      aria-labelledby="language-select"
      class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
      id="language-select"
      role="button"
      tabindex="0"
    >
      English
    </div>
    <input
      aria-hidden="true"
      class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
      tabindex="-1"
      value="en"
    />
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
      data-testid="ArrowDropDownIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
    <fieldset
      aria-hidden="true"
      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        class="css-hdw1oc"
      >
        <span
          class="notranslate"
        >
          ​
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Header renders Header ThemePicker 1`] = `
<div>
  <div
    class="theme-picker"
  >
    <div
      class="MuiToggleButtonGroup-root css-1gjgmky-MuiToggleButtonGroup-root"
      data-testid="select-theme"
      role="group"
    >
      <button
        aria-pressed="false"
        class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-primary MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-d9c359-MuiButtonBase-root-MuiToggleButton-root"
        tabindex="0"
        type="button"
        value="girly"
      >
        <img
          alt="theme logo"
          class="box-image"
          src="colorMode.png"
        />
        <div
          class="img-back"
        />
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        aria-pressed="false"
        class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-primary MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-d9c359-MuiButtonBase-root-MuiToggleButton-root"
        tabindex="0"
        type="button"
        value="light"
      >
        <img
          alt="theme logo"
          class="box-image"
          src="dayMode.png"
        />
        <div
          class="img-back"
        />
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        aria-pressed="true"
        class="MuiButtonBase-root MuiToggleButton-root Mui-selected MuiToggleButton-sizeMedium MuiToggleButton-primary MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-d9c359-MuiButtonBase-root-MuiToggleButton-root"
        tabindex="0"
        type="button"
        value="dark"
      >
        <img
          alt="theme logo"
          class="box-image"
          src="nightMode.png"
        />
        <div
          class="img-back"
        />
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
